ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/SpectraS3PaginationLoader_Test.java

```diff
@@ -0,0 +1,109 @@
+ */
+
+package com.spectralogic.ds3client.integration;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3client.Ds3Client;
+import com.spectralogic.ds3client.commands.spectrads3.PutBulkJobSpectraS3Request;
+import com.spectralogic.ds3client.helpers.Ds3ClientHelpers;
+import com.spectralogic.ds3client.helpers.pagination.GetObjectsFullDetailsLoaderFactory;
+import com.spectralogic.ds3client.integration.test.helpers.TempStorageIds;
+import com.spectralogic.ds3client.integration.test.helpers.TempStorageUtil;
+import com.spectralogic.ds3client.models.ChecksumType;
+import com.spectralogic.ds3client.models.DetailedS3Object;
+import com.spectralogic.ds3client.models.bulk.Ds3Object;
+import com.spectralogic.ds3client.utils.collections.LazyIterable;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.spectralogic.ds3client.integration.Util.deleteAllContents;
+import static com.spectralogic.ds3client.integration.Util.insecureFromEnv;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SpectraS3PaginationLoader_Test {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SpectraS3PaginationLoader_Test.class);
+
+    private static final Ds3Client CLIENT = Util.fromEnv();
+    private static final Ds3ClientHelpers HELPERS = Ds3ClientHelpers.wrap(CLIENT);
+    private static final String TEST_ENV_NAME = "s3_pagination_test";
+    private static final int RETRIES = 5;
+
+    private static TempStorageIds envStorageIds;
+    private static UUID envDataPolicyId;
+
+    @BeforeClass
+    public static void startup() throws IOException {
+        LOG.info("Starting test Setup...");
+        envDataPolicyId = TempStorageUtil.setupDataPolicy(TEST_ENV_NAME, false, ChecksumType.Type.MD5, CLIENT);
+        envStorageIds = TempStorageUtil.setup(TEST_ENV_NAME, envDataPolicyId, CLIENT);
+        LOG.info("Finished test Setup...");
+    }
+
+    @AfterClass
+    public static void teardown() throws IOException {
+        LOG.info("Starting test teardown...");
+        TempStorageUtil.teardown(TEST_ENV_NAME, envStorageIds, CLIENT);
+        CLIENT.close();
+        LOG.info("Finished test teardown...");
+    }
+
+    @Test
+    public void basicPagination() throws IOException {
+        try {
+            // test setup
+            HELPERS.ensureBucketExists(TEST_ENV_NAME);
+            final int numObjects = 50;
+            CLIENT.putBulkJobSpectraS3(new PutBulkJobSpectraS3Request(TEST_ENV_NAME, createTestList(numObjects)));
+
+            final GetObjectsFullDetailsLoaderFactory loaderFactory = new GetObjectsFullDetailsLoaderFactory(CLIENT, TEST_ENV_NAME, "", 10, RETRIES, false);
```

Comment:
Much better test if numObjects is not an integral multiple of page size. I think that may work today.